### ISSUE TITLE
Word wrap not working for lines above 1000 characters

### ISSUE THREAD

**Issue by adrianth** [Microsoft/vscode#21674]
- VSCode Version: 1.10.0
- OS Version: Windows 10 x64

Steps to Reproduce:

1. Create a new file containing one line that is 1001 characters long. At this point, Alt+Z switches word wrap on and off, as expected. Leave word wrap on.
2. Click on a different tab or press Ctrl+N to switch to editing a different file.
3. Go back to the original file. Word wrap is now disabled and Alt+Z doesn't do anything anymore.
4. Delete exactly one character, so that the line is 1000 characters long.
5. Switch to a different tab and back again. Word wrap is now working fine.

**Comment by mjbvz** [comment]
Can you please share your user settings. I had trouble reproducing this on my machine

**Comment by adrianth** [comment]
Here they are:
```
{
    "editor.autoClosingBrackets": false,
    "editor.fontSize": 13,
    "editor.mouseWheelZoom": true,
    "editor.renderIndentGuides": true,
    "explorer.autoReveal": false,
    "typescript.check.tscVersion": false,
    "git.confirmSync": false,
    "window.zoomLevel": 1,
    "typescript.referencesCodeLens.enabled": false,
    "java.errors.incompleteClasspath.severity": "ignore",
    "editor.codeLens": false,
    "window.newWindowDimensions": "maximized",
    "editor.wordWrap": "on",
    "workbench.iconTheme": "vs-minimal",
    "editor.minimap.enabled": true
}
```
Also, here's a short vide demonstrating the issue: [https://youtu.be/M2Lk-CLdHcE](https://youtu.be/M2Lk-CLdHcE)

**Comment by refkotay** [comment]
v1.10.1
ubuntu 16.04 64-bit
Default settings
No extensions

I'm having a similar problem. Word wrap correctly remains off when typing/editing normally. Even past 1000 chars.

But type 1001+ chars, switch tabs and back, then the text becomes wrapped to viewport. Existing files with 1001+ chars also wrap when opened.

**Comment by alexandrudima** [comment]
Duplicate of #21552.

TL;DR: We try to detect minified/generated files and enforce wrapping on them. If a file has more characters on lines longer than 1k chars than it has characters on lines shorter than 1k chars, we consider that this is a minified/generated (not source) file. For performance reasons (no virtual horizontal scrolling) we enforce viewport wrapping. In the insiders channel, there is a fix where Toggle Word Wrap can unwrap such files. Read more about it in #21552.
